common/lib/xmodule/xmodule/modulestore/split_mongo/split_draft.py

```diff
@@ -198,6 +198,13 @@ def delete_item(self, location, user_id, revision=None, **kwargs):
                 branches_to_delete = [ModuleStoreEnum.BranchName.published, ModuleStoreEnum.BranchName.draft]
             elif revision is None:
                 branches_to_delete = [ModuleStoreEnum.BranchName.draft]
+                delete_children_catogries = ['chapter', 'sequential', 'vertical']
```

Comment:
```delete_children_catogries``` are only those categories which contain only xblock children while  ```DIRECT_ONLY_CATAGORIES``` not have this .